smartlearner/optimizers/adam.py

```diff
@@ -0,0 +1,43 @@
+from collections import OrderedDict
+
+import theano
+import theano.tensor as T
+import numpy as np
+
+from ..interfaces import Optimizer
+from ..utils import sharedX
+
+class Adam(Optimizer):
```

Comment:
Which would have been as well with an iterator of the gradients and the use of the opposite gradients. I'm not even sure you'd save on LOCs.